Hometask 5 films

kv67 opened the pull request.

kv67 committed:
Добавлена отправка напоминаний с задержкой по времени.
Добавлено отображение диалогового окна при получении сообщений из Firebase

kv67 committed:
Добавлена новая таблица Notifications в БД.
Добавлены миграции.
Реализовано добавление напоминаний в список (таблицу в БД).
Реализованы функции редактирования и удаления напоминаний.

kv67 committed:
Реализовано удаление данных из таблицы notification при очистке БД (пункт меню "Очистить данные")
Доработана отправка сообщения в AlarmManager

baidala commented on app/src/main/java/kve/ru/firstproject/service/FilmNotificationPublisher.kt:
```diff
@@ -0,0 +1,114 @@
+                    val alarmSound =
+                        RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+                    setSound(alarmSound)
+                    setContentIntent(contentIntent)
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)     создаем NotificationChannel

baidala commented on app/src/main/java/kve/ru/firstproject/fragments/FilmDetailFragment.kt:
```diff
@@ -48,6 +59,37 @@ class FilmDetailFragment : Fragment() {
+            val year = cal.get(Calendar.YEAR)
+            val month = cal.get(Calendar.MONTH)
+            val day = cal.get(Calendar.DAY_OF_MONTH)
+            val dialog = DatePickerDialog(
```
нужно запросить дату и время

baidala reviewed (changes requested):
надо добавить канал для уведомлений и  дату+ время в *Посмотреть позже*

kv67 committed:
Добавлена возможность назначать время для напоминания.
Реализовано открытие детальной страницы фильма при получении сообщения из Firebase.

kv67 committed:
Доработана обработка уведомлений, полученных из Firebase, если приложение запущено и на переднем крае.
